static/src/board.js

```diff
@@ -31,13 +31,25 @@ function drawGrid(ctx) {
     ctx.stroke();
 }
 
-function drawPuzzle(ctx, board) {
+function delay(time) {
+    return new Promise(resolve => setTimeout(resolve, time));
```

Comment:
Promises are there own thing, but are basically way to set up callbacks so that they can be called in sequence and reads easily.  

In this case the Promise resolves (completes) when timeout is done.  In C we don't need anything fancy because everything in C happens in sequence, in JS we basically put work to the side to be done later so making callbacks happen in a certain order requires extra work. 

This example is a bit of an unusual usage of promises.